### PR TITLE
[Metricbeat][Kubernetes] Add test metrics family to scheduler, proxy and controller manager

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?

Currently, all `state_*` metricsets have a test that runs `TestStateMetricsFamily` in the testing file. However, controller manager, scheduler and proxy metricsets were missing that. This PR fixes that. 

This testing function was first introduced [in this PR](https://github.com/elastic/beats/pull/34448).

### CODE REVIEW

File: metricbeat/module/kubernetes/controllermanager/controllermanager_test.go
Comment (ChrsMark) [inline metricbeat/module/kubernetes/controllermanager/controllermanager_test.go:53]: That can be confusing since `controllermanager` and the rest control-plane are not retrieving metrics from kube-state-metrics. 
So I think that we should not use this helper function here. If `TestStateMetricsFamily` is generic and can be used for any kind of Prometheus metrics and not just for "state"metrics then I suggest we rename it properly and use it accordingly here too.

**Review by ChrsMark — CHANGES_REQUESTED** [review]

File: metricbeat/module/kubernetes/controllermanager/controllermanager_test.go
Comment (constanca-m) [reply metricbeat/module/kubernetes/controllermanager/controllermanager_test.go:53]: You are right @ChrsMark , I didn't notice the name had `state` in it. I will change it, since the function is generic and update this PR

Comment (constanca-m) [reply metricbeat/module/kubernetes/controllermanager/controllermanager_test.go]: I removed the `state` now in the name. Luckily the description of the function was very generic, so I updated only the name.

File: metricbeat/module/kubernetes/controllermanager/controllermanager_test.go
Comment (constanca-m) [reply metricbeat/module/kubernetes/controllermanager/controllermanager_test.go:23]: Thanks @ChrsMark , I will try, but why do you think that is the reason? I looked at it, but couldn't understand.

Comment (ChrsMark) [reply metricbeat/module/kubernetes/controllermanager/controllermanager_test.go:23]: I'm not 100% sure how this linter works but usually when I hit this kind of issues (related to imports) I usually fix them by tuning the order: https://github.com/elastic/beats/actions/runs/5923829670/job/16060200841#step:5:31

**Review by ChrsMark — APPROVED** [review]
lgtm!

**Review by tetianakravchenko — APPROVED** [review]

**Review by cmacknz — APPROVED** [review]
You should update the CODEOWNER for metricbeat/helper/kubernetes to match the module/kubernetes owner. Then you won't need agent team approval next time.

**Pull request merged**
